src/Compiler/Service/TransparentCompiler.fs

```diff
@@ -778,7 +848,7 @@ type internal TransparentCompiler
                     }
         }
 
-    let ComputeBootstrapInfo (projectSnapshot: ProjectSnapshot) =
+    let ComputeBootstrapInfo (fileName: string option) (projectSnapshot: ProjectSnapshot) =
```

Comment:
The `BootstrapInfo` shouldn't be tied to a particular file. Unless it doesn't matter for which file we do this then you might need to rethink it. 

As it stands the file name is not part of the cache key, so it will be computed with whichever file it gets called for first and then just retrieved from cache for any other file. 

If we added it to the cache key here then each file would have its own BoostrapInfo, which we probably don't want.

Reply:
Uhm, yeah sorry about that.
I am struggling with all the parameters that LoadClosure.ComputeClosureOfScriptText needs.
Here's another approach under the assumption that we only ever need the closure if `UseScriptResolutionRules = true`  and a single source file is in the project snapshot.